linux/nvme.go

```
@@ -0,0 +1,220 @@
+    }
+    return nil
+}
+
+func setSysctl(key, value string) error {
+    cmd := fmt.Sprintf("sysctl -w %s=%s", key, value)
+    out, _, err := util.ExecCommandOutput("sh", []string{"-c", cmd})
+    if err != nil {
+        return fmt.Errorf("failed to set %s: %v (%s)", key, err, out)
+    }
+    return nil
+}
+
+func setKernelParam(path, value string) error {
+    f, err := os.OpenFile(path, os.O_WRONLY, 0)
+    if err != nil {
+        return fmt.Errorf("failed to open %s: %v", path, err)
+    }
+    defer f.Close()
+    if _, err := f.WriteString(value); err != nil {
+        return fmt.Errorf("failed to write %s to %s: %v", value, path, err)
+    }
+    return nil
+}
+
+// ConnectNvmeTarget connects to an NVMe over TCP target
+func ConnectNvmeTarget(target *model.NvmeTarget) error {
+    discoveryIPs := strings.Split(target.Address, ",")
+    if len(discoveryIPs) == 0 {
+        return fmt.Errorf("no discovery IPs provided for NVMe target")
+    }
+
+    var lastErr error
+    var success bool
+
+    for _, ip := range discoveryIPs {
+        args := []string{
+            "connect",
+            "-t", "tcp",
+            "-n", target.NQN,
+            "-a", ip,
+            "-s", target.Port,
+        }
+        _, rc, err := util.ExecCommandOutput(nvmecmd, args)
+        if err != nil && rc != 114 {
+            log.Warnf("NVMe connect failed for discovery IP %s, rc=%d, error: %s", ip, rc, err)
+            lastErr = err
+            continue
+        }
+        log.Infof("Successfully connected to NVMe target using discovery IP %s", ip)
+        success = true
+    }
+
+    if !success {
+        return fmt.Errorf("failed to connect to NVMe target: %v", lastErr)
+    }
+    return nil
+}
+
+// DisconnectNvmeTarget disconnects from an NVMe target
+func DisconnectNvmeTarget(target *model.NvmeTarget) error {
+    args := []string{
+        "disconnect",
+        "-n", target.NQN,
+    }
+    
+    _, _, err := util.ExecCommandOutput(nvmecmd, args)
```

Can we log the error? Also, is it ok to ignore the error here?